packages/babylon/src/plugins/typescript.js

```diff
@@ -1704,6 +1708,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     // `let x: number;`
     parseVarHead(decl: N.VariableDeclarator): void {
       super.parseVarHead(decl);
+      if (this.eat(tt.bang)) {
```

Comment:
Like @nicolo-ribaudo, it probably doesn't make sense to parse a `!` here unless there's a way to issue an error later.